### PR TITLE
[IOTDB-3015] Authenticate login via confignode

### PR DESCRIPTION
**Pull request by RYH61**
Authenticate login via confignode

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java [hunk 1]
```diff
@@ -355,4 +356,9 @@ public DataSet queryPermission(ConfigRequest configRequest) {
       return dataSet;
     }
   }
+
+  @Override
+  public TSStatus login(String username, String password) {
+    return AuthorInfo.getInstance().login(username, password);
```
[the remark this answers is not in the document]
Comment (RYH61) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java:362]: fix

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java [hunk 2]
```diff
@@ -107,4 +108,17 @@ public SettableFuture<ConfigTaskResult> queryPermission(TAuthorizerReq authorize
...
+
+  public TSStatus login(TLoginReq req) {
+    ConfigNodeClient configNodeClient = null;
+    TSStatus status = null;
+    try {
+      configNodeClient = new ConfigNodeClient();
+      // Send request to some API server
+      status = configNodeClient.login(req);
+    } catch (IoTDBConnectionException | BadNodeUrlException e) {
+      LOGGER.error("Failed to connect to config node.");
```
[the remark this answers is not in the document]
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java:120]: TSStatus doesn't seem to support doing this

[the remark this answers is not in the document]
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java:120]: fix

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java [hunk 3]
```diff
@@ -107,4 +108,17 @@ public SettableFuture<ConfigTaskResult> queryPermission(TAuthorizerReq authorize
...
     return future;
   }
+
+  public TSStatus login(TLoginReq req) {
+    ConfigNodeClient configNodeClient = null;
+    TSStatus status = null;
+    try {
+      configNodeClient = new ConfigNodeClient();
+      // Send request to some API server
+      status = configNodeClient.login(req);
```
[the remark this answers is not in the document]
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java:119]: TSStatus doesn't seem to support doing this also

[the remark this answers is not in the document]
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/ClusterAuthorizer.java:119]: fix

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java [hunk 4]
```diff
@@ -56,6 +56,27 @@ public class AuthorInfo {
...
+      status = false;
+      loginMessage = e.getMessage();
+    }
+    if (status) {
+      tsStatus.setCode(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+      tsStatus.setMessage("Login successfully");
+    } else {
+      tsStatus.setMessage(loginMessage != null ? loginMessage : "Authentication failed.");
+      tsStatus.setCode(TSStatusCode.WRONG_LOGIN_PASSWORD_ERROR.getStatusCode());
+    }
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java:76]: Is it possible that the user doesn't exist when AuthException occurs? You can add a TSStatusCode to express this~

**Review by CRZbulabula — COMMENTED** [review]
PTAL~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java (hunk 4 above)
Comment (RYH61) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java]: I will return WRONG_LOGIN_USERNAME_OR_PASSWORD_ERROR when the user does not exist or the password is wrong

Comment (RYH61) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java:76]: WRONG_LOGIN_PASSWORD_ERROR means that the username or password is wrong

**Review by qiaojialin — APPROVED** [review]

**Review by MarcosZyk — APPROVED** [review]

**Pull request merged**